src/public/components/Menu/Menu.js

```diff
@@ -0,0 +1,76 @@
+import ButtonComponent from '../Button/Button.js';
+import HeaderComponent from '../Header/Header.js';
+
+export default class MenuComponent {
+    constructor ({
+        parent = document.body,
+        pages = {}
+    } = {}) {
+        this._parent = parent;
+        this.menuItems = {
+            singleplayer: 'Singleplayer',
+            multiplayer: 'Multiplayer',
+            score: 'Scoreboard',
+            authors: 'Authors'
+        };
+        this.pages = pages;
+        this.header = null;
+        this.mainMenuButtons = {};
+        this._elem = null;
+        this.onClickItem = this.onClickItem.bind(this);
+    }
+
+    onClickItem (event) {
+        const currentTarget = event.target;
+
+        if ((currentTarget instanceof HTMLButtonElement)) {
```

Comment:
Тоже лучше инвертировать условие и по нему выходить из функции:
```suggestion
        if (!(currentTarget instanceof HTMLButtonElement)) {
            return;
        }
```